### PR TITLE
fix: improve typing for SyncBase and AsyncBase

### PR DESCRIPTION
**Pull request by kumaraditya303**

### CODE REVIEW

File: playwright/_impl/_async_base.py [hunk 1]
```diff
@@ -15,19 +15,19 @@
...
 
 
 T = TypeVar("T")
-Self = TypeVar("Self", bound="AsyncBase")
+Self = TypeVar("Self", bound="AsyncContextManager")
 
 
 class AsyncEventInfo(Generic[T]):
-    def __init__(self, future: asyncio.Future) -> None:
+    def __init__(self, future: "asyncio.Future[T]") -> None:
```
Comment (pavelfeldman) [inline playwright/_impl/_async_base.py:30]: I don't think Future is generic in 3.7, 3.8.

Comment (pavelfeldman) [reply playwright/_impl/_async_base.py:30]: Ah, I see you are using quotes to work around

Comment (kumaraditya303) [reply playwright/_impl/_async_base.py:30]: Any other changes required here ?

**Pull request merged**